src/stirling/source_connectors/socket_tracer/socket_trace_connector.cc

```diff
@@ -653,22 +663,40 @@ void SocketTraceConnector::CheckTracerState() {
 
   int error_code;
   openssl_trace_state_->get_value(kOpenSSLTraceStatusIdx, error_code);
+  bool mismatched_fds = error_code == kOpenSSLMismatchedFDsDetected;
 
-  if (error_code == kOpenSSLMismatchedFDsDetected) {
-    openssl_trace_mismatched_fds_counter_family_.Add({{"name", openssl_mismatched_fds_metric}})
-        .Increment();
+  if (FLAGS_stirling_debug_tls_sources || mismatched_fds) {
+    if (mismatched_fds) {
+      openssl_trace_mismatched_fds_counter_family_.Add({{"name", openssl_mismatched_fds_metric}})
+          .Increment();
+    }
 
     // Record the offending applications and clear the BPF hash in the process.
     auto table = openssl_trace_state_debug_->get_table_offline(true);
     for (auto& entry : table) {
       struct openssl_trace_state_debug_t debug = std::get<1>(entry);
       auto ssl_source = std::string(magic_enum::enum_name(debug.ssl_source));
+      auto is_mismatched_entry = debug.mismatched_fd;
+
+      if (is_mismatched_entry) {
```

Comment:
Why isn't this gated on the `if (FLAGS_stirling_debug_tls_sources)` condition as well? Or should it be?

Reply:
The mismatched fd detection is important for validating a core assumption of our new tls tracing. That logic existed prior to this debug mode and we always want that to run.

This conditional logic was introduced because the debug mode and mismatched fd detection conveniently can be instrumented in the same code path. Otherwise I would have needed to add more BPF maps and scaffolding to accomplish this change.